R/as_curve_params.R

```diff
@@ -23,7 +22,10 @@ as_curve_params <- function(data, antigen_isos = NULL) {
 
   # check if object is curve (with columns)
   if (!all(is.element(curve_cols, curve_data %>% names()))) {
-    cli::cli_abort("Please provide curve data")
+    cli::cli_abort(
+      class = "not curve_params",
+      message = c("Please provide curve data.")
```

Comment:
Here are some additional improvements that we can move to a separate issue:

- [ ] Let's make this error specify which columns are missing.  Please use [command substitution](https://cli.r-lib.org/reference/inline-markup.html#command-substitution) and [inline vector collapsing](https://cli.r-lib.org/reference/inline-markup.html#collapsing-inline-vectors) to do this.

- [ ] Please review the tidyverse style guide's section on [error messages](https://style.tidyverse.org/error-messages.html) for additional instructions; please try to implement all of their recommendations, and let me know if there are any you're unsure about.

Reply:
(moved to https://github.com/UCD-SERG/serocalculator/issues/220)